partiql-parser/src/result.rs

```diff
@@ -0,0 +1,82 @@
+// Copyright Amazon.com, Inc. or its affiliates.
+
+//! [`Error`] and [`Result`] types for parsing PartiQL.
+
+use pest::error::{ErrorVariant, LineColLocation};
+use std::fmt;
+use thiserror::Error;
+
+/// Position in the source for an error.
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub enum Position {
+    Unknown,
+    At { line: usize, column: usize },
+}
```

Comment:
This variant makes it easier to deal with the case that we might not have a position.  Could've used `Option<Position>` and make this a `struct` but that makes implementing `Display` and embedding in our `ParserError` more clumsy, so I just encoded the variants here.

Reply:
This has changed a little in #12 to make the structure more explicit (but not to the degree you are suggesting).  Basically it still looks a lot like `Option`, but the main reason for the breakout was actually a friction between the `Display` implementation in the derived `Error` (via `thiserror`) and `Option<...>` which was my original variant.

The other reason why it is not `Option` is that I think we might have this be potentially a span if applicable, but that is not a given...